### ISSUE TITLE
[BUG] Error of bead normalization in data preprocessing

### ISSUE THREAD

**Issue by HaidYi** [kevin931/PyCytoData#9]
**What issues are you experiencing?**

RuntimeWarning: divide by zero encountered in divide
  data[:, np.isin(channels, c)] = data[:, np.isin(channels, c)]/(correct_f_full.reshape(data.shape[0], 1)) #type: ignore
RuntimeWarning: invalid value encountered in divide
  data[:, np.isin(channels, c)] = data[:, np.isin(channels, c)]/(correct_f_full.reshape(data.shape[0], 1)) #type: ignore

**To Reproduce**
It's hard to reproduce since it's because of the randomness from `np.empty`. To reproduce it, you can try the pipeline on a dataset, and you will find after every run of the preprocessing with bead normalization, the `n_cells` is different.

**Expected behavior**
The expected behavior is to have reproducible preprocessing pipeline.

**Your environment:**
 - OS: MacOS 12.6
 - Python Version: 3.10
 - PyCytoData version: 0.1.0
 - PyCytoData distribution: PyPi

**Additional context**

After checking the source code, it seems that the variable `interval_mean` is reset before computing the correlation in the `bead_normalization` function.  You can refer to this line: https://github.com/kevin931/PyCytoData/blob/5f8ba0e43be571ffad66ec8f518034169a36bd2a/PyCytoData/preprocess.py#L293

**Comment by kevin931** [comment]
Thanks for letting me know! I tried it out with a dataset, and indeed ``interval_mean`` is the culprit. It was intended that we need to find the interval twice: once before applying the quantile-based filter and once after. However, the bug is that the step of finding the mean was omitted the second time, resulting in some random stuff in that array. Here is a snippet of what it should have:

```python
  interval_mean: np.ndarray = np.empty((unique_intervals.shape[0], data.shape[1]), dtype=float) # type: ignore
  for i in range(0, unique_intervals.shape[0]):
      interval_mean[i] = np.mean(data[np.isin(intervals, unique_intervals[i]),:], axis=0)
```
Regarding the fact that sample size decreases after each run, I think this is the correct behavior since the bead normalization algorithm has a quantile-based filtering that I previously mentioned. I will double check this as well.

Anyway, **I will fix this and expedite the release of `0.1.1`.** Thanks again! Hope this addresses the issue that you're running into!

**Comment by kevin931** [comment]
I've published a new release (v0.1.1) with a patch to this issue. It is available both on PyPI and conda. I will close this issue at some point if all is good. But if you have more questions or if the issue persists, feel free to chime in here and let me know!
